File: WIT.Portal/src/app/entities/product-class.entity.ts
import { BaseEntity } from './base.entity';
import { ListBase, GridInfo } from './grid-info.entity';

export class ProductClass extends BaseEntity {
    public productClassID: number;
    public code: string;
    public name: string;
    public description: string;

    constructor()
    constructor(rhs: ProductClass)
    constructor(rhs?: ProductClass) {
        super(rhs);
        if (!rhs) {
            this.productClassID = 0;
        }
    }

    public get caption() {
        return this.code + '-' + this.name || '';
    }

    public get isValid() {
        return ((this.code || "") !== "");
    }
}

export class ProductClassList extends ListBase {
    public items: Array<ProductClass>;
    public code: string;
    public name: string;

    constructor()
    constructor(rhs: ProductClassList)
    constructor(rhs?: ProductClassList) {

        super(rhs);

        if (!this.items || !Array.isArray(this.items)) {
            this.items = new Array<ProductClass>();
        } else {
            this.items = this.items.map(v => new ProductClass(v));
        }
    }
}

File: WIT.Business/Transactions/ProductClassEntity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Collections;
using WIT.Data.Models;

namespace WIT.Business.Entities
{
    public class ProductClassItem : BaseEntity
    {
        public int ProductClassID { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }

        public ProductClassItem() { }
        public ProductClassItem(ProductClass rhs)
        {
            CreatedOn = rhs.CreatedOn;
            UpdatedOn = rhs.UpdatedOn;
            CreatedBy = rhs.CreatedBy;
            UpdatedBy = rhs.UpdatedBy;

            ProductClassID = rhs.ProductClassID;
            Code = rhs.Code;
            Name = rhs.Name;
            Description = rhs.Description;
        }
    }

    public class ProductClassList
    {
        public GridInfo GridInfo { get; set; }
        public List<ProductClassItem> Items { get; set; }

        public string Code { get; set; }
        public string Name { get; set; }

        public ProductClassList()
        {
            Items = new List<ProductClassItem>();
            GridInfo = new GridInfo();
        }
    }
}
